packages/ui/src/hooks/useTable/useTable.ts

```diff
@@ -174,15 +181,17 @@ export default function useTable<T extends GenericRowStructure>({
     const handleRowsPerPageChange = (event: ChangeEvent<HTMLInputElement>) => {
         const updatedRowsPerPage = parseInt(event.target.value, 10);
         updateState({ rowsPerPage: updatedRowsPerPage, page: 0 });
+        onChangeRowsPerPage?.(updatedRowsPerPage);
     };
 
     const handleSortCellClick = (cellId: CellId<T>) => {
         if (state.sortByColumnId === cellId) {
-            updateState({
-                sortDirection: state.sortDirection === 'asc' ? 'desc' : 'asc',
-            });
+            const sortDirection = state.sortDirection === 'asc' ? 'desc' : 'asc';
+            updateState({ sortDirection: sortDirection });
+            onClickSortCell?.(cellId, sortDirection);
         } else {
             updateState({ sortByColumnId: cellId, sortDirection: 'asc' });
```

Comment:
if sorting is controlled,no need to update the state. see `handleRowSelection`. maybe a better way to do this is to check if the functionality (either sorting, filtering or selection) is controlled, then, from the hook return back the handlers passed as props, otherwise return the handlers that update the internal state